roles/matrix-mxisd/tasks/setup_mxisd.yml

```diff
@@ -22,7 +23,29 @@
     source: "{{ 'pull' if ansible_version.major > 2 or ansible_version.minor > 7 else omit }}"
     force_source: "{{ matrix_mxisd_docker_image_force_pull if ansible_version.major > 2 or ansible_version.minor >= 8 else omit }}"
     force: "{{ omit if ansible_version.major > 2 or ansible_version.minor >= 8 else matrix_mxisd_docker_image_force_pull }}"
-  when: matrix_mxisd_enabled|bool
+  when: matrix_mxisd_enabled|bool and not matrix_mxisd_self_build
+
+- name: Ensure gradel is installed for self-building
+  apt:
+    name:
+      - gradle
+    state: present
+    update_cache: yes    
+  when: "matrix_mxisd_enabled|bool and matrix_mxisd_self_build"
+
+- name: Ensure mxisd repository is present on self-build
+  git:
+    repo: https://github.com/kamax-matrix/mxisd.git
+    dest: "{{ matrix_mxisd_docker_src_files_path }}"
+    version: "v{{ matrix_mxisd_docker_image.split(':')[1] }}"
+    force: "yes"
+  when: "matrix_mxisd_enabled|bool and matrix_mxisd_self_build"
+
+- name: Ensure mxisd Docker image is build
+  shell: "./gradlew dockerBuild"
```

Comment:
I wonder if there's a way to avoid running gradle and just invoke `docker build ..` directly. Well, via the `docker_image` Ansible module, like we do for the others.

But I guess you've already toyed with that and it's not as easy..

Reply:
Unfortunately I haven't found any way. :/

Reply:
Indeed.. It looks like building the image relies on copying the jar file into it.

The actual building of the jar file happens outside and requires pulling in dependencies, all done using `gradle`.